core/controller/cluster/controller.go

```diff
@@ -103,76 +104,80 @@ type Controller interface {
 	GetClusterBuildStatus(ctx context.Context, clusterID uint) (*BuildStatusResponse, error)
 	GetResourceTree(ctx context.Context, clusterID uint) (*GetResourceTreeResponse, error)
 	GetStep(ctx context.Context, clusterID uint) (resp *GetStepResponse, err error)
+	// Deprecated: for internal usage, v1 to v2
+	Upgrade(ctx context.Context, clusterID uint) error
```

Comment:
If it is a interface for internal usage, how about make it a stand alone application?

Reply:
The interface is coupled to the core code and needs to exist for a long time. So how about marking it deprecated for easier maintenance?